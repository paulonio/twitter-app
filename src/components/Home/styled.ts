import styled from 'styled-components';

export const HomeWrapper = styled.section`
  width: 57.5%;
  max-width: 910px;
  border-left: solid 1px ${({ theme }) => theme.colors.border};
  border-right: solid 1px ${({ theme }) => theme.colors.border};

  @media screen and (max-width: 1480px) {
    width: 70%;
  }

  @media screen and (max-width: 768px) {
    width: 100%;
  }
`;

export const HeaderSpace = styled.div`
  width: 100%;
  height: 58px;
`;
