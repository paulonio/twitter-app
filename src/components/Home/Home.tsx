import React from 'react';
import { useSelector } from 'react-redux';
import { Navigate } from 'react-router-dom';
import { HeaderSpace, HomeWrapper } from './styled';
import Header from '../Header/Header';
import TweetBlock from '../TweetBlock/TweetBlock';
import Tweets from '../Tweets/Tweets';
import { StoreType } from '../../store';
import type { User } from '../../store/slices/authSlice';

const Home = () => {
  const user = useSelector<StoreType, User | null>((state) => state.auth.user);

  if (!user) {
    return <Navigate to="/" replace />;
  }

  return (
    <HomeWrapper>
      <Header>Home</Header>
      <HeaderSpace />
      <TweetBlock />
      <Tweets type="feed" />
    </HomeWrapper>
  );
};

export default Home;
