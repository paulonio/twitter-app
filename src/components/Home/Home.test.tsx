import React from 'react';
import { render } from '@testing-library/react';

import Home from '@components/Home/Home';
import CustomProvider from '../EditUser/testUtils';

describe('Home', () => {
  it('should render home component', () => {
    const { container } = render(
      <CustomProvider>
        <Home />
      </CustomProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
