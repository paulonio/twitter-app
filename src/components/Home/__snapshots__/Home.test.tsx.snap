// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render home component 1`] = `<div />`;
